packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -154,7 +154,7 @@ export const useGridFilter = (apiRef: GridApiRef, rowsProp: GridRowsProp): void
         }
 
         if (newItem.id == null) {
-          newItem.id = new Date().getTime();
+          newItem.id = Math.random();
```

Comment:
I had a similar case for the Infinite loader feature. `Math.random()` returns a number between `[0, 1]`. I think its better to multiply it by something to have it be an `Int`.

Reply:
The `useId` hook uses `Math.round(Math.random() * 1e5)`.

Reply:
I was thinking about this as well, and wanted to introduce uuid. RandomInt is not ideal 
Also, not sure why the useId is a hook.